src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -133,8 +198,7 @@ public static Vector128<byte> ShiftRightBytesInVector(Vector128<byte> value, [Co
             return AdvSimd.ExtractVector128(value, Vector128<byte>.Zero, numBytes);
         }
 
-        ThrowUnreachableException();
-        return default;
+        return Vector128.Shuffle(value, Vector128.Create((byte)0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15) + Vector128.Create(numBytes));
```

Comment:
In .NET 9+ the `Vector128.Create((byte)0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15)` can be `Vector128<byte>.Indices`

The total `Vector128.Create((byte)0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15) + Vector128.Create(numBytes)` can itself become `Vector128.CreateSequence<byte>(start: numBytes, step: 1)`

Reply:
.NET 8 only unfortunately. Good to see a new API though, writing these is cumbersome and not great to read.